File: CPP_programming/Day6_class with ptr/Book/Book.cpp
#include"Book.h"

int Book::cnt=0;
Book::Book()
{
    cnt++;
    book_id= cnt;
    book_name= new char[3];
    strcpy(book_name,"NA");
}
Book::Book(const char *s, float price)
{
    cnt++;
    book_id = cnt;
    this->book_name=new char[(strlen(s))+1];
    strcpy(book_name,s);
    this->price = price;
}
Book::Book(Book &obj)
{
    cnt++;
    this->book_id = cnt;
    this->book_name = new char[(strlen(obj.book_name))+1];
    strcpy(this->book_name, obj.book_name);
    this->price = obj.price;
}
Book::~Book()
{
    delete[] book_name;
}

void Book::display()
{
    cout<<"Book ID: "<<book_id<<"\n";
    cout<<"Name: "<<book_name<<"\n";
    cout<<"Price: "<<price<<"\n";
}

File: Core_java_new/Day3_OOP in java 3/A3_Book/Book.java
class Book
{
	private int bookid;
	private static int cnt;
	private float price;
	
	{
		cnt++;
	}
	
	Book()
	{
		bookid=cnt;
	}
	Book(float price)
	{
		bookid=cnt;
		this.price=price;
	}
	
	public void display()
	{
		System.out.println("Book ID: "+bookid);
		System.out.println("Price: "+price);
	}
	
	
}